### PR TITLE
OOP School Library: Decorators

### PR DESCRIPTION
**Pull request by AmanWorku**
## Activities done:
- [x] Created an interface called `Nameable`.
- [x] Made the `Person` class inherit from `Nameable`.
- [x] Created a base decorator called `Decorator`.
- [x] Implemented `CapitalizeDecorator` and `TrimmerDecorator` used for capitalizing the given word and trimming the word if its length is greater than 10 respectively.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -18,6 +22,10 @@ def of_age?
   def can_use_services?
     of_age? || @parent_permission
   end
+
+  def correct_name
+    @name.capitalize
+  end
 end
 
 private :of_age?
```
Comment (gemmen29) [inline person.rb:31]: - [ ] Kindly move this line `private :of_age?` inside the class as it's now unrelated to the class.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @AmanWorku, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
@gemmen29 Thank you, I will be correcting it accordingly.

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @AmanWorku 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer
- Excellent communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
@MahmoudBakr23 Thank you so much.

**Pull request merged**